datatypes.h

```diff
@@ -550,12 +550,13 @@ typedef enum {
 	PPM_CTRL_TYPE_CURRENT,
 	PPM_CTRL_TYPE_CURRENT_NOREV,
 	PPM_CTRL_TYPE_CURRENT_NOREV_BRAKE,
+	PPM_CTRL_TYPE_CURRENT_BRAKE_REV_HYST,
+	PPM_CTRL_TYPE_CURRENT_SMART_REV,
 	PPM_CTRL_TYPE_DUTY,
 	PPM_CTRL_TYPE_DUTY_NOREV,
 	PPM_CTRL_TYPE_PID,
 	PPM_CTRL_TYPE_PID_NOREV,
-	PPM_CTRL_TYPE_CURRENT_BRAKE_REV_HYST,
-	PPM_CTRL_TYPE_CURRENT_SMART_REV
```

Comment:
It is best to not move enums around as that will break old configs. Always add new enums to the end. This is why the order of the enums doesn't always make sense.

Reply:
I was thinking since each config data structure has a unique hash and all settings are erased on firmware upload, that that wouldn't be a problem. But I guess you store configs by enum value instead of by name on the vesc tool side.
```
<app_ppm_conf.ctrl_type>0</app_ppm_conf.ctrl_type>
```

Reply:
For the serialization it does, but not when you load a config stored as xml. Then every enum is stored by value.

Reply:
Reverted the enum reordering.